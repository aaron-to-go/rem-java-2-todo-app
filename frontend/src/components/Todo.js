import styled from 'styled-components/macro'

import { Button, CardContent } from '@material-ui/core'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardMedia from '@material-ui/core/CardMedia'
import makeStyles from '@material-ui/core/styles/makeStyles'
import { Link } from 'react-router-dom'

const useStyles = makeStyles(() => ({
  media: {
    height: 0,
    paddingTop: '56.25%',
  },
}))

export default function Todo({ todo, onDelete, onAdvance, detailView }) {
  const classes = useStyles()

  return (
    <Card>
      <CardMedia
        image="https://cdn.shibe.online/shibes/8c463b877b520f4e5de642d0b94325b7e9d3e262.jpg"
        title="Paella dish"
        className={classes.media}
      />

      <CardContent>
        <Description>Description: {todo.description}</Description>
      </CardContent>
      {!detailView && (
        <CardActions>
          <Button color="secondary" onClick={() => onDelete(todo)}>
            Delete
          </Button>

          <Button color="primary" component={Link} to={`/todo/${todo.id}`}>
            Details
          </Button>

          {onAdvance && (
            <Button color="primary" onClick={() => onAdvance(todo)}>
              Advance
            </Button>
          )}
        </CardActions>
      )}
    </Card>
  )
}

const Description = styled.span`
  grid-column: span 2;
  font-weight: 600;
`
