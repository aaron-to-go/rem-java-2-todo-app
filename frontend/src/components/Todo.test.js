import renderer from 'react-test-renderer';
import Todo from "./Todo";

it("snapshot test todo", () => {
    //GIVEN
    const todo = {
        id: "1234",
        description: "Hallo"
    }

    //WHEN
    const tree = renderer.create(<Todo todo={todo}
                                       onAdvance={() => console.log("Advanced")}
                                       onDelete={() => console.log("On Delete")}
    />).toJSON();

    //THEN
    expect(tree).toMatchSnapshot();
});
