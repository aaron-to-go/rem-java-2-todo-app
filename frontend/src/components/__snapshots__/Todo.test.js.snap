// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test todo 1`] = `
<section
  className="Todo__Wrapper-sc-1u29sm1-0 hwKVuW"
>
  <span
    className="Todo__Description-sc-1u29sm1-1 bkfpda"
  >
    Description: 
    Hallo
  </span>
  <button
    className="sc-bdfBwQ dxYAvz"
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    className="sc-bdfBwQ iZlgSj"
    onClick={[Function]}
  >
    Advance
  </button>
</section>
`;
